### PR TITLE
Added slicing functions in BWR files

### PR DESCRIPTION
**Pull request by igorcantele**
Added `_Slice` and `_ChannelSlicer _TimeSlicer _Slicer` classes.  

**tests**  
Added `tests.test_slicing` testing file.

**docs**
Updated docs.

### CODE REVIEW

File: bwpy/__init__.py [hunk 1]
```diff
@@ -30,9 +31,11 @@ def _establish_type(self):
         if self.description.startswith("BRW"):
             self.__class__ = BRWFile
             self._type = "brw"
+            self.__post_init__()
```
Comment (Helveg) [inline bwpy/__init__.py:34]: Is `__post_init__` defined anywhere?

File: bwpy/__init__.py [hunk 2]
```diff
@@ -107,12 +115,122 @@ def guid(self):
...
+    def __getitem__(self, instruction):
+        slice = self._slice._time_slice(instruction)
+        #self._file.n_frames = 
+        return slice
+
+
+class _ChannelSlicer(_Slicer):
+    def __getitem__(self, instruction):
+        slice = self._slice._channel_slice(instruction)
+        #self._file.n_channel =
```
Comment (Helveg) [inline bwpy/__init__.py:133]:
```suggestion
```

**Review by Helveg — CHANGES_REQUESTED** [review]
Looks great!

File: setup.py [hunk 3]
```diff
@@ -24,6 +24,6 @@
         "Programming Language :: Python :: 3",
         "Operating System :: OS Independent",
     ],
-    install_requires=["h5py<3.0.0", "numpy"],
+    install_requires=["h5py", "numpy"],
```
[the remark this answers is not in the document]
Comment (Igor10798) [reply setup.py:27]: I wanted to check it out but when I try to run `python -m unittest tests.test_file.TestUserInfo.test_raw_user_info` to check if the test is okay from the `bwpy` folder I receive the following error:  
```
File "path_to_module\bwpy\tests\test_file.py", line 4, in <module>
    from helpers import get_sample_path, open_sample, open_sample_copy, samples
ModuleNotFoundError: No module named 'helpers'
```

File: bwpy/__init__.py (hunk 1 above)
Comment (Igor10798) [reply bwpy/__init__.py:34]: Line 210  
```
def __post_init__(self):
        _Slice.__init__(self, self)
```

**Comment by igorcantele** [comment]
Let my contribution become live :(

File: docs/source/index.rst [hunk 4]
```diff
@@ -31,6 +31,47 @@ BWR and BXR files can be opened as a regular :class:`h5py.File` objects (see `Fi
...
+-------
+
+The package allows you to slice the data in `.brw` files. The data can be restricted to certain time samples by indexing the `.t` property like a one-dimensional array:
+.. code-block:: python
+
+   import bwpy
+
+   with bwpy("my_data.bwr", "r") as datafile:
+      datafile.t[0:10:2]
+      #will return the slice of the first 10 temporal recordings with a step of 2
```
Comment (Helveg) [inline docs/source/index.rst:44]:
```suggestion
      # Return the slice of the first 10 temporal recordings with a step of 2
      datafile.t[0:10:2]
```

File: docs/source/index.rst [hunk 5]
```diff
@@ -31,6 +31,47 @@ BWR and BXR files can be opened as a regular :class:`h5py.File` objects (see `Fi
...
+
+The data can be restricted to certain channels by indexing the `.ch` property like a two-dimensional array:
+
+.. code-block:: python
+
+   import bwpy
+
+   with bwpy("my_data.bwr", "r") as datafile:
+      datafile.ch[0:10, 0: 10]
+      #will return the slice of the block of the first 10x10 channels
```
Comment (Helveg) [inline docs/source/index.rst:54]:
```suggestion
      # Return the slice of the block of the first 10x10 channels
      datafile.ch[0:10, 0: 10]
```

File: docs/source/index.rst [hunk 6]
```diff
@@ -31,6 +31,47 @@ BWR and BXR files can be opened as a regular :class:`h5py.File` objects (see `Fi
...
+
+The obtained slices can themselves be sliced further:
+
+.. code-block:: python
+
+   import bwpy
+
+   with bwpy("my_data.bwr", "r") as datafile:
+      datafile.t[0:10].ch[0, 0]
+      #will return the slice of the first 10 temporal recordings of the first channel
```
Comment (Helveg) [inline docs/source/index.rst:64]:
```suggestion
      # Return the slice of the first 10 temporal recordings of the first channel
      datafile.t[0:10].ch[0, 0]
```

**Review by Helveg — CHANGES_REQUESTED** [review]

**Review by Helveg — APPROVED** [review]

**Pull request merged**